prediction-polls/android/app/src/main/java/com/bounswe/predictionpolls/ui/login/LoginScreenViewModel.kt

```diff
@@ -0,0 +1,48 @@
+package com.bounswe.predictionpolls.ui.login
+
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.navigation.NavController
+import com.bounswe.predictionpolls.core.BaseViewModel
+import com.bounswe.predictionpolls.data.remote.repositories.AuthRepository
+import com.bounswe.predictionpolls.ui.feed.navigateToFeedScreen
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+
+@HiltViewModel
+class LoginScreenViewModel @Inject constructor(
+    private val authRepository: AuthRepository
+) : BaseViewModel() {
+    var screenState by mutableStateOf(LoginScreenState())
+        private set
+
+    fun onEvent(event: LoginScreenEvent) {
+        screenState = screenState.reduce(event)
+
+        when (event) {
+            is LoginScreenEvent.OnLoginButtonClicked -> onLoginButtonClicked(event.navController)
+            is LoginScreenEvent.OnLoginWithGoogleButtonClicked -> onLoginWithGoogleButtonClicked()
+            else -> {}
+        }
+    }
+
+    private fun onLoginButtonClicked(navController: NavController) {
```

Comment:
VMs should not interact with the UI this way, VM is responsible for creating state. I dont think passing UI elements to the view model is a good practice.

Reply:
Should i pass it as a closure? I think you re right but I dont know the answer

Reply:
Ideally, you could set a variable like shouldNavigate in the state holder class, but this is fine too i reconsidered